parlai/core/hugging_face_bpe_helper.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from parlai.core.opt import Opt
+from typing import List
+
+
+class HuggingFaceBpeHelper(object):
+    @staticmethod
+    def add_cmdline_args(argparser):
+        parser = argparser.add_argument_group('HuggingFaceBpeHelper Arguments')
+        parser.add_argument(
+            '--bpe-vocab', type=str, help='path to pre-trained tokenizer vocab'
+        )
+        parser.add_argument(
+            '--bpe-merge', type=str, help='path to pre-trained tokenizer merge'
+        )
+        return parser
+
+    def __init__(self, opt: Opt, shared=None):
+        try:
+            from tokenizers import ByteLevelBPETokenizer
+        except ImportError:
+            raise ImportError(
+                'Please install huggingface tokenizer with: pip install tokenizers'
+            )
+
+        if opt.get('bpe_vocab', None) is None:
+            raise ValueError('--bpe-vocab is required for loading pretrained tokenizer')
+        if opt.get('bpe_merge', None) is None:
+            raise ValueError('--bpe-merge is required for loading pretrained tokenizer')
+        self.vocab_path = opt.get('bpe_vocab')
+        self.merge_path = opt.get('bpe_merge')
+        self.tokenizer = ByteLevelBPETokenizer(self.vocab_path, self.merge_path)
+
+    def encode(self, text: str) -> List[str]:
+        return self.tokenizer.encode(text).tokens
+
+    def decode(self, x: List[str]) -> str:
+        return self.tokenizer.decode(self.tokenizer.token_to_id(c) for c in x)
```

Comment:
are you sure that you need to call token_to_id individually here?

Reply:
It depends on how the decoding interface works, our dict expects a list of tokens, while the hugging face expects a list of token ids. 
What do you suggest here?

Reply:
I think this is good, we can look for an improvement later if we think it's a bottleneck.